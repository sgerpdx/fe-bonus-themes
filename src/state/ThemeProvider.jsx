import React, { createContext, useState, useContext } from 'react';
//import PropTypes from 'prop-types';

export const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('space');

  const handleThemeChange = (e) => {
    setTheme((theme) => (theme = e.target.value));
  };

  const state = { theme, handleThemeChange };

  return (
    <ThemeContext.Provider value={state}>{children}</ThemeContext.Provider>
  );
};

export const useTheme = () => {
  const { theme } = useContext(ThemeContext);
  return theme;
};

export const useThemeChange = () => {
  const { handleThemeChange } = useContext(ThemeContext);
  return handleThemeChange;
};

export default ThemeProvider;
