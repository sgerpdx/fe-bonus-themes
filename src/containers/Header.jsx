import React from 'react';
import ThemeControl from '../components/controls/ThemeControl';
import styles from '../components/app/App.css';
import { useTheme } from '../state/ThemeProvider';

export default function Header() {
  const theme = useTheme();
  return (
    <div
      className={`${
        theme === 'earth'
          ? styles.headerThemeEarth
          : theme === 'vulcan'
          ? styles.headerThemeVulcan
          : styles.headerThemeSpace
      }`}
    >
      <ThemeControl />
    </div>
  );
}
