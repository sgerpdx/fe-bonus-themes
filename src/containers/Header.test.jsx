/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Header from './Header';
import ThemeProvider from '../state/ThemeProvider';

describe('header container component', () => {
  it('displays a heading and a drop-down selection menu', async () => {
    render(
      <ThemeProvider>
        <Header />
      </ThemeProvider>
    );

    const controlInstructions = await screen.getByText(
      'Use the menu below to select a visual theme:'
    );
    expect(controlInstructions).toMatchSnapshot();
  });
});
