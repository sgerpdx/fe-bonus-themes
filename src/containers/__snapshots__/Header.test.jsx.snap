// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header container component displays a heading and a drop-down selection menu 1`] = `
<h4>
  Use the menu below to select a visual theme:
</h4>
`;
